### ISSUE TITLE
Silent failure when filename includes path to a directory that does not exist.

### ISSUE THREAD

**Issue by vsbuffalo** [rgreinho/trauma#73]
Thanks for writing trauma! It's a super nice library that works very well. I noticed this behavior that may or not be a bug, but it is subtle and could trip folks up.

# Bug Report

When manually creating a `Download` entry, if `filename` contains an output path to a directory that does not exist, trauma will not provide an error, nor automatically create the directory; instead, it will silently not do anything.

## Possible Solution

Probably easiest to just raise an `Err()` and let upstream users handle directory creation. But it could also have a `bool` that would automatically create the output directories and continue on with downloads.

## Example

Here is a small MRE: 

```rust
use std::path::PathBuf;
use trauma::{download::Download, downloader::DownloaderBuilder, Error};

use reqwest::Url;

#[tokio::main]
async fn main() -> Result<(), Error> {
    let reqwest_rs = "https://github.com/seanmonstar/reqwest/archive/refs/tags/v0.11.9.zip";
    let downloads = vec![Download { url: Url::parse(reqwest_rs).unwrap(), filename: "test_dir/reqwest.zip".to_string() } ];
    let downloader = DownloaderBuilder::new()
        .build();
    downloader.download(&downloads).await;
    Ok(())
}
```

`Cargo.toml`:

```rust
[package]
name = "trauma_mre"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
reqwest = "0.11.20"
tokio = "1.32.0"
trauma = "2.2.3"
```

Let me know if you need more info — I am fairly new to Rust, so unsure if other info is needed for a full bug report.

**Comment by rgreinho** [comment]
Thanks a lot for the report! 

Indeed, I think that is something I did not take into account. I'll check to see what makes more sense, either creating the path with the file, either returning an error. But I agree it should definitely not fail silently!

**Comment by rgreinho** [comment]
The fix is available in [v2.2.4](https://github.com/rgreinho/trauma/releases/tag/2.2.4).

Thanks again for the great bug report.
